pkg/apis/monitoring/v1beta1/alertmanager_config_types.go

```diff
@@ -565,6 +567,26 @@ type HTTPConfig struct {
 	FollowRedirects *bool `json:"followRedirects,omitempty"`
 }
 
+// WebexConfig configures notification via Cisco Webex
+// See https://prometheus.io/docs/alerting/latest/configuration/#webex_config
+type WebexConfig struct {
+	// The Webex Teams API URL i.e. https://webexapis.com/v1/messages
+	// +optional
+	APIURL string `json:"apiURL,omitempty"`
+	// The HTTP client's configuration.
+	// You must use this configuration to supply the bot token as part of the HTTP `Authorization` header.
+	// +optional
+	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
+	// Message template
+	// +optional
+	Message string `json:"message,omitempty"`
+	// ID of the Webex Teams room where to send the messages.
+	RoomID string `json:"roomID"`
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+}
```

Comment:
I've changed the optional fields to pointers and accommodated the `convertWebexConfig` function to work with the new definition.

Reply:
I've also changed the internal `webexConfig` struct in the `alertmanager` package to use pointers. 

E2E tests are still failing. The YAML config can't be parsed for the following reason:

```
ts=2023-06-17T06:10:30.122Z caller=coordinator.go:118 level=error component=configuration msg="Loading configuration file failed" file=/etc/alertmanager/config_out/alertmanager.env.yaml err="missing webex_configs.http_config.authorization"
```

Why is this happening? The e2e tests for webex were written in a similar fashion to the telegram ones, and the e2e were passing before moving to pointers.

Reply:
OK... I needed to take a look on how the global config for the tests is being created, then everything was clear. I had to make some changes:

I had to change the HTTPConfig field from optional to mandatory, as the Bearer Token of the Bot is expected to be there. The AlertManager code always checks for its presence:

https://github.com/prometheus/alertmanager/blob/90a8acd27661db09c9d9f1b4a2d4574a24466c76/config/notifiers.go#L196-L213
```go
// UnmarshalYAML implements the yaml.Unmarshaler interface.
func (c *WebexConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
	*c = DefaultWebexConfig
	type plain WebexConfig
	if err := unmarshal((*plain)(c)); err != nil {
		return err
	}

	if c.RoomID == "" {
		return fmt.Errorf("missing room_id on webex_config")
	}

	if c.HTTPConfig == nil || c.HTTPConfig.Authorization == nil {
		return fmt.Errorf("missing webex_configs.http_config.authorization")
	}

	return nil
}
``` 

I decided against expecting from the user to put the Authorization for the webex bot into their global config and made the http_config a mandatory field.

After that, the tests were still failing, because of missing smpt fields in the global AlertManager config. I've added an smpt_smarthost and a smpt_from, which are also mandatory fields for a global config (the config couldn't be unmarshalled for else wise).

Finally, I noticed that the `testPromOperatorStartsWithoutScrapeConfigCRD` was still failing. The test was checking whether the returned map with pod restarts of the prometheus operator pod was empty:

```goland
	require.Emptyf(t, restarts, "expected to get 1 container but got %d", len(restarts))
// this produced the funny looking error: expected to get 1 container but got 1
```

I've changed this to check whether only 1 container was really returned.

Sorry the long comment; just wanted to make sure I've explained all the changes.